docroot/themes/contrib/civictheme/includes/utilities.inc

```diff
@@ -415,3 +420,110 @@ function civictheme_get_entity_field_key_values(EntityInterface $entity, $field_
 
   return $values;
 }
+
+/**
+ * Check if the provided link is external.
+ *
+ * @param string $url
+ *   The url to be validated.
+ * @param string $host
+ *   The current site host.
+ * @param array $overridden_domains
+ *   Array of overridden domains.
+ *
+ * @return bool
+ *   TRUE if the link is external, FALSE otherwise.
+ */
+function civictheme_link_is_external(string $url, string $host, array $overridden_domains) {
+  // If the link does not have a domain.
+  if ((strpos($url, 'internal:') === 0) || (strpos($url, '/') === 0)) {
+    return FALSE;
+  }
+  // If the link has a domain.
+  $url_data = parse_url($url);
+  if (!empty($url_data['host'])) {
+    $domain = trim($url_data['host'], '/');
+    // If domain is the same as the current site.
+    if ($domain == $host) {
+      return FALSE;
+    }
+    // Included in the list of the overridden domains, including wildcards.
+    if (!empty($overridden_domains)) {
+      foreach ($overridden_domains as $overridden_domain) {
+        if (strpos($overridden_domain, $domain) !== FALSE) {
+          return FALSE;
+        }
+      }
+    }
+  }
+  return TRUE;
+}
+
+/**
+ * Get the list of override domains.
+ *
+ * @return array
+ *   List of override domains.
+ */
+function civictheme_get_override_domains() {
+  $config_factory = \Drupal::service('config.factory');
+  $theme_name = $config_factory->get('system.theme')->get('default');
+  $config = $config_factory->get("$theme_name.settings");
+  $overridden_domains = $config->get('civictheme_override_domains') ?? '';
+  if (!empty($overridden_domains)) {
+    return explode("\r\n", $overridden_domains);
+  }
+  return [];
+}
+
+/**
+ * Adds CSS class to all links.
+ *
+ * @param string $string
+ *   The string to be processed.
+ *
+ * @return string
+ *   The processed string.
+ */
+function civictheme_process_link_attributes(string $string) {
+  $dom = Html::load($string);
```

Comment:
afaik, this may throw exceptions on the invalid HTML

could you please validate this and, if it is a case - please wrap this processing into `try/catch` and catch on exception and return original `$html` as is (basically - return broken HTML back).

Reply:
Does not give exceptions. 
![Screenshot 2022-08-19 at 3 26 38 PM](https://user-images.githubusercontent.com/83997348/185594441-f09ff146-8086-4c7a-ad71-126c9f6b5e4c.png)
There is an issue for it to provide exceptions in future but as of not not 
https://www.drupal.org/project/drupal/issues/2950595